File: src/0754/S0754_v1.py
class Solution:
    def reachNumber(self, target: int) -> int:
        return n if (t:=abs(target)) == (s:=(n:=round((2*t)**0.5))*(n+1)>>1) or s-t&1 == 0 else n+1+(n&1)

File: src/0754/S0754_v1.java
class Solution {
    public int reachNumber(int target) {
        if (target < 0) target = -target;
        int n = (int)(Math.sqrt(target<<1) + 0.5);
        int s = n * n + n >> 1;
        return ((s - target & 1) == 0  || s == target) ? n : n + 1 + (n&1);
    }
}
